### PR TITLE
fix(l1): re-validate blob sidecars against the payload fork when building

### PR DESCRIPTION
**Pull request by ElFantasma**
## Motivation

Closes the EF finding `ethrex-getpayload-v5-legacy-proof`. A blob transaction's sidecar is validated against the fork **only at mempool insertion** (`add_blob_transaction_to_pool`). The payload builder then fetches the stored bundle by hash and aggregates it with **no fork-aware re-check**, and `GetPayloadV5Request::handle` gates only on the Osaka *timestamp* before serializing the bundle as-is.

So if a **pre-Osaka v0 bundle** (EIP-4844, 1 proof/blob) survives in the mempool across Osaka activation and is picked by the first post-Osaka build, `getPayloadV5` emits a `BlobsBundleV2` carrying 1 proof/blob instead of the 128 cell proofs EIP-7594/Osaka requires — a wrong-format Engine-API payload. There is no v0→cell-proof upgrade path, so the correct behavior is to not include such a tx.

## Description

In `Blockchain::apply_blob_transaction`, after fetching the sidecar, **re-validate it against the fork being built** (`context.chain_config().fork(header.timestamp)`) via the existing `BlobsBundle::validate_cheap`, and skip the tx (the builder's standard `EvmError::Custom` skip signal, as used for "max data blobs reached") if it's incompatible. The stale tx is left in the mempool and simply not included in a post-Osaka payload.

- Reuses `validate_cheap` (which already rejects a v0 bundle at `fork >= Osaka`) — no new validation logic, no cell-proof conversion.
- Gated on `#[cfg(feature = "c-kzg")]` like the rest of blob validation (`validate_cheap` is itself `c-kzg`-only; without the feature there is no sidecar cryptography to validate). Verified both feature combos build.
- Explicit builds (`testing_buildBlockV1`) carry no sidecar (`bundle` is `None`), so this is a no-op for them.

## On testing

No synthetic test is added, deliberately:
- The trigger is a **transient activation-boundary** condition (a v0 sidecar surviving the mempool across the Osaka fork and being selected by the first post-Osaka build). Hive / execution-spec-tests / ef_tests exercise steady-state behavior on each side of a fork, not mempool carry-over *across* activation — which is why the finding is tagged **needs-devnet**. A devnet crossing the boundary is the real validation; no CI stage reproduces it.
- The reused check, `validate_cheap` rejecting a v0 bundle at `fork >= Osaka`, is **already unit-tested** (`test/tests/common/blobs_bundle_tests.rs`, from the #6768 blob-constraints work). A new test would only exercise the wiring, at the cost of full KZG-blob + payload-build setup for a needs-devnet, medium-leaning-low finding.

## Checklist

- [x] Builds with **and** without `c-kzg`; `cargo clippy -p ethrex-blockchain --features c-kzg` clean; existing `payload_tests` + `explicit_blob_tx_tests` pass (no regression to normal blob building).

### CODE REVIEW

File: crates/blockchain/payload.rs [hunk 1]
```diff
@@ -920,6 +920,25 @@ impl Blockchain {
...
             }
         };
+        // A blob sidecar is validated against the fork only at mempool insertion; the fork being
+        // built can differ (e.g. Osaka activated since). Re-validate against the payload's fork so
+        // a stale sidecar is skipped rather than served: a pre-Osaka v0 bundle (1 proof/blob) has
+        // no cell-proof upgrade path and is invalid post-Osaka, so including it would make
+        // getPayloadV5 emit a wrong-format BlobsBundleV2. (Explicit builds carry no sidecar, so
+        // `bundle` is `None` and this is a no-op.) Gated on `c-kzg` like the rest of blob
+        // validation — without it there is no sidecar cryptography to validate against.
+        #[cfg(feature = "c-kzg")]
```
[the remark this answers is not in the document]
Comment (ElFantasma) [reply crates/blockchain/payload.rs]: Fixed in 300145528 — dropped the `#[cfg(feature = "c-kzg")]` gate. The check is now **structural** (`bundle.version == 0 && chain_config.is_osaka_activated(ts)`): `version` is a plain `u8` and `is_osaka_activated` needs no KZG, so it holds in every build. It no longer routes through the c-kzg-only `validate_cheap`.

**Comment by ElFantasma** [comment]
Thanks — reworked the fix per the reviews (`300145528`):

**@greptile (P1, feature gate):** dropped the `c-kzg` gate; the check is now structural (`bundle.version == 0 && is_osaka_activated(timestamp)`) and applies in all builds — no dependency on the c-kzg-only `validate_cheap`.

**@Codex #2 / @Kimi (permanent invalidity → hard-drop):** a post-Osaka v0 sidecar is deterministically and permanently invalid, so it's now a **hard drop** — `self.remove_transaction_from_pool(&tx_hash)` before the skip, exactly like the frame-tx fork gates. It no longer lingers as a retry-every-block execution failure that blocks later nonces from the same sender or spams failure metrics.

**@Codex #1 (other serving paths):** the hard-drop removes the tx from the mempool, so it's no longer served by P2P pooled-tx reads either (not just excluded from `getPayloadV5`). The remaining read path — `engine_getBlobsV2/V3` — is tracked as its **own** finding, `ethrex-getblobs-v2-legacy-proof`, so it's intentionally out of this PR's scope; a proposing node also evicts the tx here at the first post-Osaka build.

**@Claude / testing:** still no synthetic test, deliberately (as discussed): the trigger is a transient activation-boundary condition no CI stage reproduces (needs-devnet), and the check is now trivial/structural. Verified builds **with and without** `c-kzg`, clippy clean, and existing `payload_tests` / `explicit_blob_tx_tests` / `get_payload_v5_*` pass (no regression; the empty-bundle explicit-build case confirms the `Some(bundle)` guard leaves explicit builds untouched).

**Review by ilitteri — APPROVED** [review]

**Review by MegaRedHand — APPROVED** [review]

**Review by pablodeymo — APPROVED** [review]
